setup_libuv.py

```diff
@@ -1,257 +1,188 @@
+    ]
+elif sys.platform == 'darwin':
+    SOURCES += [
+        'deps/libuv/src/unix/bsd-ifaddrs.c',
+        'deps/libuv/src/unix/darwin.c',
+        'deps/libuv/src/unix/darwin-proctitle.c',
+        'deps/libuv/src/unix/fsevents.c',
+        'deps/libuv/src/unix/kqueue.c',
+        'deps/libuv/src/unix/proctitle.c',
+        'deps/libuv/src/unix/pthread-barrier.c',
+    ]
+elif sys.platform.startswith('freebsd'):
+    SOURCES += [
+        'deps/libuv/src/unix/bsd-ifaddrs.c',
+        'deps/libuv/src/unix/freebsd.c',
+        'deps/libuv/src/unix/kqueue.c',
+        'deps/libuv/src/unix/posix-hrtime.c',
+    ]
+elif sys.platform.startswith('dragonfly'):
+    SOURCES += [
+        'deps/libuv/src/unix/bsd-ifaddrs.c',
+        'deps/libuv/src/unix/freebsd.c',
+        'deps/libuv/src/unix/kqueue.c',
+        'deps/libuv/src/unix/posix-hrtime.c',
+    ]
+elif sys.platform.startswith('openbsd'):
+    SOURCES += [
+        'deps/libuv/src/unix/bsd-ifaddrs.c',
+        'deps/libuv/src/unix/kqueue.c',
+        'deps/libuv/src/unix/openbsd.c',
+        'deps/libuv/src/unix/posix-hrtime.c',
+    ]
+elif sys.platform.startswith('netbsd'):
+    SOURCES += [
+        'deps/libuv/src/unix/bsd-ifaddrs.c',
+        'deps/libuv/src/unix/kqueue.c',
+        'deps/libuv/src/unix/netbsd.c',
+        'deps/libuv/src/unix/posix-hrtime.c',
+    ]
+
+elif sys.platform.startswith('sunos'):
+    SOURCES += [
+        'deps/libuv/src/unix/no-proctitle.c',
+        'deps/libuv/src/unix/sunos.c',
+    ]
 
 
 class libuv_build_ext(build_ext):
-    libuv_dir      = os.path.join('deps', 'libuv')
-    libuv_repo     = 'https://github.com/libuv/libuv.git'
-    libuv_branch   = 'v1.x'
-    libuv_revision = 'd6ac141'
-    libuv_patches  = []
+    libuv_dir = os.path.join('deps', 'libuv')
 
     user_options = build_ext.user_options
     user_options.extend([
-        ("libuv-clean-compile", None, "Clean libuv tree before compilation"),
-        ("libuv-force-fetch", None, "Remove libuv (if present) and fetch it again"),
-        ("libuv-verbose-build", None, "Print output of libuv build process"),
-        ("use-system-libuv", None, "Use the system provided libuv, instead of the bundled one")
+        ("use-system-libuv", None, "Use the system provided libuv, instead of the bundled one"),
     ])
     boolean_options = build_ext.boolean_options
-    boolean_options.extend(["libuv-clean-compile", "libuv-force-fetch", "libuv-verbose-build", "use-system-libuv"])
+    boolean_options.extend(["use-system-libuv"])
 
     def initialize_options(self):
         build_ext.initialize_options(self)
-        self.libuv_clean_compile = 0
-        self.libuv_force_fetch = 0
-        self.libuv_verbose_build = 0
         self.use_system_libuv = 0
 
     def build_extensions(self):
-        self.force = self.force or self.libuv_force_fetch or self.libuv_clean_compile
-
         if self.compiler.compiler_type == 'mingw32' or self.use_system_libuv:
```

Comment:
OK I think figured it out. Previously, you couldn't build bundled libuv because we used vcbuild.bat (which is using VS compiler). 

So I guess this is no longer necessary.